utils/git.go

```diff
@@ -154,6 +155,64 @@ func (gm *GitManager) Checkout(branchName string) error {
 	return nil
 }
 
+func (gm *GitManager) CheckoutToHash(hash, targetBranchWd string) error {
+	log.Debug("Running git fetch...")
+	if err := gm.Fetch(); err != nil {
+		return err
+	}
+	log.Debug("Running git checkout to hash:", hash)
+	if err := gm.createBranchAndCheckoutToHash(hash, false); err != nil {
+		return fmt.Errorf("'git checkout %s' failed with error: %s", hash, err.Error())
+	}
+	return nil
+}
+
+func (gm *GitManager) Fetch() error {
+	log.Debug("Running git fetch...")
+	err := gm.localGitRepository.Fetch(&git.FetchOptions{
+		RemoteName: gm.remoteName,
+		RemoteURL:  gm.remoteGitUrl,
+		Auth:       gm.auth,
+	})
+	if err != nil && err != git.NoErrAlreadyUpToDate {
```

Comment:
what happens when the error is NoErrAlreadyUpToDate? will the fetch is still performed when we got this error?

Reply:
the fetch fails since there is nothing to fetch, in this case, we want to continue, any other error will be returned.